src/translator/response_builder.h

```diff
@@ -81,7 +82,7 @@ class ResponseBuilder {
   // Data members are context/curried args for the functor.
 
   ResponseOptions responseOptions_;
-  std::vector<Ptr<Vocab const>> *vocabs_; // vocabs are required for decoding
+  Vocabs *vocabs_; // vocabs are required for decoding
```

Comment:
Minor && sorry to bother, but... Is there any reason why a `unique_ptr` is not used here?

Reply:
Lifetime/ownership is handled at Service. This is a reference alive so long as Service is.

Reply:
In this case you need to use `Vocab& vocabs_` or to be more precise `const Vocab& vocabs_`. Using raw pointers in C++ in 2021 is only really justifiable if you are implementing some very low level datastructures. I have refactored the code.

Reply:
(Correct me if I am wrong)

Reply:
@XapaJIaMnu Thanks for refactoring. We certainly do not want users to change the vocabs members outside this class.

Reply:
Ok, seriously, we use Vocab * in a lot of places, I'm really not sure why, This would warrant further refactoring. Can someone explain to me the logic behind it. Is it because we want to move objects a lot and we want full move semantics?

Reply:
@XapaJIaMnu Thanks. I didn't know of this syntax (most C++ I learned was pre 2011). That's why there's Vocab* in places. I'll update the rest in a following PR.

Reply:
@jerinphilip @XapaJIaMnu  I will do the rest refactorings by replacing `Vocabs *` to `const Vocabs&`.

Reply:
In Google style it's & for const and * for mutable but I tend to just use & for everything.